### PR TITLE
Add UIKit equivalent APIs for animating implicitly.

### PR DESCRIPTION
**Pull request by jverkoey**
This will enable easier migrations from existing UIKit code paths while allowing clients to benefit from the increased support for implicitly animatable properties.

### CODE REVIEW

File: src/MDMMotionAnimator.h [hunk 1]
```diff
@@ -167,6 +167,130 @@ NS_SWIFT_NAME(MotionAnimator)
...
+ @param animations From UIKit's documentation: "A block object containing the changes to commit to
+ the views. This is where you programmatically change any animatable properties of the views in your
+ view hierarchy. This block takes no parameters and has no return value. This parameter must not be
+ NULL." Supports animating additional CALayer properties beyond what UIView's similar API supports.
+ See MDMAnimatableKeyPaths for a full list of implicilty animatable CALayer properties.
+ @param completion From UIKit's documentation: "A block object to be executed when the animation
+ sequence ends. This block has no return value and takes a single Boolean argument that indicates
+ whether or not the animations actually finished before the completion handler was called." Unlike
+ UIKit's API, if the duration of the animation is 0, this block is performed immediately. This
+ parameter may be NULL.
```
Comment (romoore) [inline src/MDMMotionAnimator.h]: Nit: This huge block of text is hard to parse. Maybe format to make reading the headers a little easier?

```
@param completion From UIKit's documentation: "A block object to be executed when the animation
                  sequence ends. This block has no return value and takes a single Boolean 
                  argument that indicates whether or not the animations actually finished before 
                  the completion handler was called." Unlike UIKit's API, if the duration of the 
                  animation  is 0, this block is performed immediately. This parameter may be NULL.
```

**Review by romoore — APPROVED** [review]
Looks good, just a nit about the documentation formatting.

File: src/MDMMotionAnimator.h (hunk 1 above)
Comment (jverkoey) [reply src/MDMMotionAnimator.h]: Done.
